tools/Environments/DevHome.Environments/ViewModels/OperationsViewModel.cs

```diff
@@ -18,26 +28,46 @@ public enum OperationKind
 /// Represents an operation that can be performed on a compute system.
 /// This is used to populate the launch and dot buttons on the compute system card.
 /// </summary>
-public partial class OperationsViewModel
+public partial class OperationsViewModel : IEquatable<OperationsViewModel>
 {
+    private readonly Guid _operationId = Guid.NewGuid();
+
     private readonly OperationKind _operationKind;
 
+    private readonly string _additionalContext = string.Empty;
+
     public string Name { get; }
 
-    public string? Description { get; set; }
+    public ComputeSystemOperations ComputeSystemOperation { get; }
 
     public string IconGlyph { get; }
 
     private Func<string, Task<ComputeSystemOperationResult>>? ExtensionTask { get; }
 
     private Action? DevHomeAction { get; }
 
-    public OperationsViewModel(string name, string icon, Func<string, Task<ComputeSystemOperationResult>> command)
+    public OperationsViewModel(string name, string icon, Func<string, Task<ComputeSystemOperationResult>> command, ComputeSystemOperations computeSystemOperation)

```

Comment:
Could have combined this and the other constructor with a parameter default value.